merino/jobs/wikipedia_indexer/filemanager.py

```diff
@@ -0,0 +1,138 @@
+from google.cloud.storage.fileio import BlobReader, BlobWriter
+
+logger = logging.getLogger(__name__)
+
+
+class DirectoryParser(HTMLParser):
+    """Parse the directory listing to find the specified file."""
+
+    filter: Pattern[str]
+    file_paths: list[str]
+
+    def __init__(self, filter_wildcard: Pattern[str]) -> None:
+        super().__init__()
+        self.file_paths = []
+        self.filter = re.compile(filter_wildcard)
+
+    def _is_href(self, k: str, v: str) -> bool:
+        return k == "href" and re.search(self.filter, v) is not None
+
+    def handle_starttag(self, tag: str, attrs: list[tuple[str, Optional[str]]]) -> None:
+        """When the parser encounters a start tag check for Anchor and push into list."""
+        if tag.lower() == "a":
+            hrefs = [(k, v) for k, v in attrs if v is not None and self._is_href(k, v)]
+            self.file_paths.extend([v for _, v in hrefs])
+
+
+class FileManager:
+    """Tools for managing files on wikimedia export directory and copying into GCS"""
+
+    base_url: str
+    gcs_bucket: str
+    object_prefix: str
+    file_pattern: Pattern
+    client: Client
+
+    def __init__(self, gcs_bucket: str, gcs_project: str, export_base_url: str) -> None:
+        self.file_pattern = re.compile(r".*/?enwiki-(\d+)-cirrussearch-content.json.gz")
+        self.client = Client(gcs_project)
+        self.base_url = export_base_url
+        if "/" in gcs_bucket:
+            self.gcs_bucket, self.object_prefix = gcs_bucket.split("/", 1)
+        else:
+            self.gcs_bucket = gcs_bucket
+            self.object_prefix = ""
+
+    def get_latest_dump(self, latest_gcs: Blob) -> Optional[str]:
+        """Find the latest export that's older than the latest on gcs."""
+        resp = requests.get(self.base_url)
+        parser = DirectoryParser(self.file_pattern)
+        parser.feed(str(resp.content))
+        links = parser.file_paths
+        if len(links) == 1:
+            name = links[0]
+            url = urljoin(self.base_url, name)
+            last_gcs_date = self._parse_date(str(latest_gcs.name))
+            link_date = self._parse_date(name)
+            if last_gcs_date < link_date:
+                return url
+        return None
+
+    def _parse_date(self, filename: str) -> dt:
+        """Parse datestring out of filename"""
+        date_match = re.match(self.file_pattern, filename)
+        if date_match:
+            try:
+                return dt.strptime(date_match.group(1), "%Y%m%d")
+            except ValueError:
+                pass
+        # return a zero date if nothing is found
+        return dt(1, 1, 1)
+
+    def get_latest_gcs(self) -> Blob:
+        """Find the most recent file on GCS"""
+        bucket = self.client.bucket(self.gcs_bucket)
+        blobs: list[Blob] = [b for b in bucket.list_blobs(prefix=self.object_prefix)]
+        blobs.sort(key=lambda b: self._parse_date(str(b.name)))
+        return blobs[-1]
```

Comment:
Just in case that the bucket is empty.
```suggestion
def get_latest_gcs(self) -> Optional[Blob]:
    ...
    return blobs[-1] if blobs else None
```